test/integration/node-specific/bson-options/raw.test.ts

```diff
@@ -0,0 +1,121 @@
+
+            const db = client.db('raw_bson_option_db', passOptionTo === 'db' ? option : undefined);
+            const collection = db.collection(
+              'raw_bson_option_coll',
+              passOptionTo === 'collection' ? option : undefined
+            );
+
+            const insertResult = await collection.insertOne({ myData: 23 }).catch(error => error);
+            const findOneResult = await collection
+              .findOne({ myData: 23 }, passOptionTo === 'operation' ? option : undefined)
+              .catch(error => error);
+
+            if (passOptionTo === 'client') {
+              // TODO(NODE-3946): When the raw option is passed to the client it crashed parsing hellos
+              // since they are returned as buffers and not js objects
+              expect(insertResult).to.be.instanceOf(MongoCompatibilityError);
+            } else {
+              expect(insertResult).to.have.property('insertedId').that.is.instanceOf(ObjectId);
+              expect(findOneResult).to.be.instanceOf(Buffer);
+            }
+          } finally {
+            await client.close();
+          }
+        });
+      }
+    });
+
+    describe('returns shared buffer', () => {
+      let client: MongoClient;
+      let collection: Collection<{ _id: number; myData: string }>;
+      beforeEach(async function () {
+        client = this.configuration.newClient();
+        collection = client.db('test_raw').collection('test_raw');
+        await collection.drop().catch(() => null);
+        await collection.insertOne({ _id: 1, myData: 'hello' });
+        await collection.insertOne({ _id: 2, myData: 'bye bye' });
+      });
+
+      afterEach(async function () {
+        await client?.close();
+      });
+
+      it('returned Buffer should not overwrite previously returned Buffer', async () => {
+        const resultOne = (await collection.findOne(
+          { _id: 1 },
+          { raw: true }
+        )) as unknown as Buffer;
+        expect(resultOne).to.be.instanceOf(Buffer);
+        expect(resultOne.indexOf(Buffer.from('hello'))).to.be.greaterThan(5);
+
+        const resultTwo = (await collection.findOne(
+          { _id: 2 },
+          { raw: true }
+        )) as unknown as Buffer;
+
+        expect(resultTwo).to.be.instanceOf(Buffer);
+        expect(resultTwo.indexOf(Buffer.from('bye bye'))).to.be.greaterThan(5);
+
+        // From the Node.js Docs for allocUnsafeSlow:
+        // However, in the case where a developer may need to retain a small chunk of memory from a pool for an indeterminate amount of time,
+        // it may be appropriate to create an un-pooled Buffer instance using Buffer.allocUnsafeSlow() and then copying out the relevant bits.
+
+        // The following expectation is mostly always true however it is possible that our
+        // call to allocUnsafe does not use the same pooled buffer depending on the allocation size
+        // expect(resultOne.buffer).to.equal(resultTwo.buffer);
```

Comment:
Note from virtual office. This test does not pass 100% of the time, most notably on windows. Leaving this here as a comment to show the expected behaviour.